From 470552eec39f44be3eb66e76aa6246e88ff47dee Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:13:09 -0400
Subject: [powerpc] powernv: Disable subcore for UP configs

Message-id: <1410545655-205645-561-git-send-email-dzickus@redhat.com>
Patchwork-id: 94441
O-Subject: [RHEL7 PATCH 560/626] powerpc/powernv : Disable subcore for UP configs
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit ad417330628c60e83692211e7161eb49e1843318
Author: Shreyas B. Prabhu <shreyas@linux.vnet.ibm.com>
Date:   Fri Jun 6 15:52:51 2014 +0530

    powerpc/powernv : Disable subcore for UP configs

    Build throws following errors when CONFIG_SMP=n
    arch/powerpc/platforms/powernv/subcore.c: In function 'cpu_update_split_mode':
    arch/powerpc/platforms/powernv/subcore.c:274:15: error: 'setup_max_cpus' undeclared (first use in this function)
    arch/powerpc/platforms/powernv/subcore.c:285:5: error: lvalue required as left operand of assignment

    'setup_max_cpus' variable is relevant only on SMP, so there is no point
    working around it for UP. Furthermore, subcore itself is relevant only
    on SMP and hence the better solution is to exclude subcore.o and
    subcore-asm.o for UP builds.

    Signed-off-by: Shreyas B. Prabhu <shreyas@linux.vnet.ibm.com>
    Reviewed-by: Srivatsa S. Bhat <srivatsa.bhat@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/Makefile b/arch/powerpc/platforms/powernv/Makefile
index 4ad0d34..d55891f 100644
--- a/arch/powerpc/platforms/powernv/Makefile
+++ b/arch/powerpc/platforms/powernv/Makefile
@@ -1,9 +1,9 @@
 obj-y   += setup.o opal-takeover.o opal-wrappers.o opal.o opal-async.o
 obj-y   += opal-rtc.o opal-nvram.o opal-lpc.o opal-flash.o
 obj-y   += rng.o opal-elog.o opal-dump.o opal-sysparam.o opal-sensor.o
-obj-y   += opal-msglog.o subcore.o subcore-asm.o
+obj-y   += opal-msglog.o
 
-obj-$(CONFIG_SMP) += smp.o
+obj-$(CONFIG_SMP) += smp.o subcore.o subcore-asm.o
 obj-$(CONFIG_PCI) += pci.o pci-p5ioc2.o pci-ioda.o
 obj-$(CONFIG_EEH) += eeh-ioda.o eeh-powernv.o
 obj-$(CONFIG_PPC_SCOM) += opal-xscom.o
-- 
1.7.1